### ISSUE TITLE
Minecolonies starts spamming a log after upgrade to 4000+

### ISSUE THREAD

**Issue by XerorBattler** [ldtteam/minecolonies#1467]
NPC stop working and log is spammed with exceptions (stacktrace show minecolonies source)

### Minecolonies version

version 1.12.1-0.8.4026 on forge-1.12.1-14.22.0.2462 (same problem on forge-1.12.1-14.22.1.2485,
on minecolonies-universal-1.12.1-0.8.3861 it works fine)

### Expected behavior

Not spamming a log

### Actual behaviour

Spamming a log a lot with

```
Condition check for target com.minecolonies.coremod.entity.ai.util.AITarget@4f43b559 threw an exception:
java.lang.ClassCastException: com.minecolonies.coremod.colony.buildings.BuildingLumberjack cannot be cast to com.minecolonies.coremod.colony.buildings.BuildingBaker
        at com.minecolonies.coremod.entity.ai.citizen.baker.EntityAIWorkBaker.getOwnBuilding(EntityAIWorkBaker.java:508) ~[EntityAIWorkBaker.class:?]
        at com.minecolonies.coremod.entity.ai.citizen.baker.EntityAIWorkBaker.getOwnBuilding(EntityAIWorkBaker.java:41) ~[EntityAIWorkBaker.class:?]
        at com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic.lambda$new$1(AbstractEntityAIBasic.java:158) ~[AbstractEntityAIBasic.class:?]
        at com.minecolonies.coremod.entity.ai.util.AITarget.test(AITarget.java:114) ~[AITarget.class:?]
        at com.minecolonies.coremod.entity.ai.basic.AbstractAISkeleton.checkOnTarget(AbstractAISkeleton.java:165) ~[AbstractAISkeleton.class:?]
        at java.util.stream.MatchOps$1MatchSink.accept(Unknown Source) [?:1.8.0_144]
        at java.util.ArrayList$ArrayListSpliterator.tryAdvance(Unknown Source) [?:1.8.0_144]
        at java.util.stream.ReferencePipeline.forEachWithCancel(Unknown Source) [?:1.8.0_144]
        at java.util.stream.AbstractPipeline.copyIntoWithCancel(Unknown Source) [?:1.8.0_144]
        at java.util.stream.AbstractPipeline.copyInto(Unknown Source) [?:1.8.0_144]
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source) [?:1.8.0_144]
        at java.util.stream.MatchOps$MatchOp.evaluateSequential(Unknown Source) [?:1.8.0_144]
        at java.util.stream.MatchOps$MatchOp.evaluateSequential(Unknown Source) [?:1.8.0_144]
        at java.util.stream.AbstractPipeline.evaluate(Unknown Source) [?:1.8.0_144]
        at java.util.stream.ReferencePipeline.anyMatch(Unknown Source) [?:1.8.0_144]
        at com.minecolonies.coremod.entity.ai.basic.AbstractAISkeleton.func_75246_d(AbstractAISkeleton.java:129) [AbstractAISkeleton.class:?]
        at net.minecraft.entity.ai.EntityAITasks.func_75774_a(SourceFile:129) [xf.class:?]
        at net.minecraft.entity.EntityLiving.func_70626_be(EntityLiving.java:763) [vq.class:?]
        at net.minecraft.entity.EntityLivingBase.func_70636_d(EntityLivingBase.java:2346) [vp.class:?]
        at net.minecraft.entity.EntityLiving.func_70636_d(EntityLiving.java:577) [vq.class:?]
        at net.minecraft.entity.EntityAgeable.func_70636_d(EntityAgeable.java:178) [vd.class:?]
        at com.minecolonies.coremod.entity.EntityCitizen.func_70636_d(EntityCitizen.java:1167) [EntityCitizen.class:?]
        at net.minecraft.entity.EntityLivingBase.func_70071_h_(EntityLivingBase.java:2166) [vp.class:?]
        at net.minecraft.entity.EntityLiving.func_70071_h_(EntityLiving.java:295) [vq.class:?]
        at net.minecraft.world.World.func_72866_a(World.java:1984) [amu.class:?]
        at net.minecraft.world.WorldServer.func_72866_a(WorldServer.java:833) [oo.class:?]
        at net.minecraft.world.World.func_72870_g(World.java:1946) [amu.class:?]
        at net.minecraft.world.World.func_72939_s(World.java:1754) [amu.class:?]
        at net.minecraft.world.WorldServer.func_72939_s(WorldServer.java:614) [oo.class:?]
        at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:765) [MinecraftServer.class:?]
        at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:396) [nz.class:?]
        at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:666) [MinecraftServer.class:?]
        at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:524) [MinecraftServer.class:?]
        at java.lang.Thread.run(Unknown Source) [?:1.8.0_144]
```
and

` Pausing Entity Lily Q. Atherton (com.minecolonies.coremod.job.baker) at BlockPos{x=87, y=63, z=74} for 0 Seconds because of error:
[20:49:04] [Server thread/INFO] [STDERR]: [com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic:onException:210]: java.lang.ClassCastException: com.minecolonies.coremod.colony.buildings.BuildingLumberjack cannot be cast to com.minecolonies.coremod.colony.buildings.BuildingBaker
[20:49:04] [Server thread/INFO] [STDERR]: [com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic:onException:210]:     at com.minecolonies.coremod.entity.ai.citizen.baker.EntityAIWorkBaker.getOwnBuilding(EntityAIWorkBaker.java:508)
[20:49:04] [Server thread/INFO] [STDERR]: [com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic:onException:210]:     at com.minecolonies.coremod.entity.ai.citizen.baker.EntityAIWorkBaker.getOwnBuilding(EntityAIWorkBaker.java:41)
[20:49:04] [Server thread/INFO] [STDERR]: [com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic:onException:210]:     at com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic.initSafetyChecks(AbstractEntityAIBasic.java:279)
[20:49:04] [Server thread/INFO] [STDERR]: [com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic:onException:210]:     at com.minecolonies.coremod.entity.ai.util.AITarget.apply(AITarget.java:125)
[20:49:04] [Server thread/INFO] [STDERR]: [com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic:onException:210]:     at com.minecolonies.coremod.entity.ai.basic.AbstractAISkeleton.applyTarget(AbstractAISkeleton.java:202)
[20:49:04] [Server thread/INFO] [STDERR]: [com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic:onException:210]:     at com.minecolonies.coremod.entity.ai.basic.AbstractAISkeleton.checkOnTarget(AbstractAISkeleton.java:176)
[20:49:04] [Server thread/INFO] [STDERR]: [com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic:onException:210]:     at java.util.stream.MatchOps$1MatchSink.accept(Unknown Source)
[20:49:04] [Server thread/INFO] [STDERR]: [com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic:onException:210]:     at java.util.ArrayList$ArrayListSpliterator.tryAdvance(Unknown Source)
[20:49:04] [Server thread/INFO] [STDERR]: [com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic:onException:210]:     at java.util.stream.ReferencePipeline.forEachWithCancel(Unknown Source)
[20:49:04] [Server thread/INFO] [STDERR]: [com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic:onException:210]:     at java.util.stream.AbstractPipeline.copyIntoWithCancel(Unknown Source)
[20:49:04] [Server thread/INFO] [STDERR]: [com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic:onException:210]:     at java.util.stream.AbstractPipeline.copyInto(Unknown Source)
[20:49:04] [Server thread/INFO] [STDERR]: [com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic:onException:210]:     at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
[20:49:04] [Server thread/INFO] [STDERR]: [com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic:onException:210]:     at java.util.stream.MatchOps$MatchOp.evaluateSequential(Unknown Source)
[20:49:04] [Server thread/INFO] [STDERR]: [com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic:onException:210]:     at java.util.stream.MatchOps$MatchOp.evaluateSequential(Unknown Source)
[20:49:04] [Server thread/INFO] [STDERR]: [com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic:onException:210]:     at java.util.stream.AbstractPipeline.evaluate(Unknown Source)
[20:49:04] [Server thread/INFO] [STDERR]: [com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic:onException:210]:     at java.util.stream.ReferencePipeline.anyMatch(Unknown Source)
[20:49:04] [Server thread/INFO] [STDERR]: [com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic:onException:210]:     at com.minecolonies.coremod.entity.ai.basic.AbstractAISkeleton.func_75246_d(AbstractAISkeleton.java:129)
[20:49:04] [Server thread/INFO] [STDERR]: [com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic:onException:210]:     at net.minecraft.entity.ai.EntityAITasks.func_75774_a(SourceFile:129)
[20:49:04] [Server thread/INFO] [STDERR]: [com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic:onException:210]:     at net.minecraft.entity.EntityLiving.func_70626_be(EntityLiving.java:763)
[20:49:04] [Server thread/INFO] [STDERR]: [com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic:onException:210]:     at net.minecraft.entity.EntityLivingBase.func_70636_d(EntityLivingBase.java:2346)
[20:49:04] [Server thread/INFO] [STDERR]: [com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic:onException:210]:     at net.minecraft.entity.EntityLiving.func_70636_d(EntityLiving.java:577)
[20:49:04] [Server thread/INFO] [STDERR]: [com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic:onException:210]:     at net.minecraft.entity.EntityAgeable.func_70636_d(EntityAgeable.java:178)
[20:49:04] [Server thread/INFO] [STDERR]: [com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic:onException:210]:     at com.minecolonies.coremod.entity.EntityCitizen.func_70636_d(EntityCitizen.java:1167)
[20:49:04] [Server thread/INFO] [STDERR]: [com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic:onException:210]:     at net.minecraft.entity.EntityLivingBase.func_70071_h_(EntityLivingBase.java:2166)
[20:49:04] [Server thread/INFO] [STDERR]: [com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic:onException:210]:     at net.minecraft.entity.EntityLiving.func_70071_h_(EntityLiving.java:295)
[20:49:04] [Server thread/INFO] [STDERR]: [com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic:onException:210]:     at net.minecraft.world.World.func_72866_a(World.java:1984)
[20:49:04] [Server thread/INFO] [STDERR]: [com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic:onException:210]:     at net.minecraft.world.WorldServer.func_72866_a(WorldServer.java:833)
[20:49:04] [Server thread/INFO] [STDERR]: [com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic:onException:210]:     at net.minecraft.world.World.func_72870_g(World.java:1946)
[20:49:04] [Server thread/INFO] [STDERR]: [com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic:onException:210]:     at net.minecraft.world.World.func_72939_s(World.java:1754)
[20:49:04] [Server thread/INFO] [STDERR]: [com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic:onException:210]:     at net.minecraft.world.WorldServer.func_72939_s(WorldServer.java:614)
[20:49:04] [Server thread/INFO] [STDERR]: [com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic:onException:210]:     at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:765)
[20:49:04] [Server thread/INFO] [STDERR]: [com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic:onException:210]:     at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:396)
[20:49:04] [Server thread/INFO] [STDERR]: [com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic:onException:210]:     at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:666)
[20:49:04] [Server thread/INFO] [STDERR]: [com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic:onException:210]:     at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:524)
[20:49:04] [Server thread/INFO] [STDERR]: [com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIBasic:onException:210]:     at java.lang.Thread.run(Unknown Source)`

### Steps to reproduce the problem

Start server and wait for world load

**Comment by Raycoms** [comment]
Yes we need to send this fix to 1.12 yet.

2017-09-20 19:57 GMT+01:00 VaclavHaramule <notifications@github.com>:

**Comment by Raycoms** [comment]
Should be fixed by now with the newest update
